js/ui/html_editor/modules/toolbar.js

```diff
@@ -503,20 +503,29 @@ if(Quill) {
                     continue;
                 }
 
-                this._markActiveFormatWidget(formatName, formatWidget, formats);
+                this._markActiveFormatWidget(formatName, formatWidget, formats, selection);
             }
 
             this._toggleClearFormatting(hasFormats || selection.length > 1);
         }
 
-        _markActiveFormatWidget(name, widget, formats) {
+        _markActiveFormatWidget(name, widget, formats, selection) {
             if(this._isColorFormat(name)) {
                 this._updateColorWidget(name, formats[name]);
             }
 
             if('value' in widget.option()) {
                 this._setValueSilent(widget, formats[name]);
             } else {
+                if(name === 'link' && selection?.length === 0) { // T1157840
```

Comment:
This is really specific code. Let's move it to a separate function?
like
```javascript
if('value' in widget.option()) {
    this._setValueSilent(widget, formats[name]);
} else if (name === 'link') {
    this._updateLinkButton();
} else {
    widget.$element().addClass(ACTIVE_FORMAT_CLASS);
}
```

I also recommend u to write a more precise note in `_updateLinkButton` function to explain why we cannot make it active. To give a link for `descendant` method and so on

Reply:
Moved to separated method
Note added
Note approved by Vlada